lib/api/bucketUpdateQuota.js

```diff
@@ -67,6 +67,7 @@ function bucketUpdateQuota(authInfo, request, log, callback) {
                 'updateBucketQuota');
             return callback(err, err.code, corsHeaders);
         } else {
+            monitoring.numberOfBucketQuota.inc();
```

Comment:
But cloudserver is stateless, so once it restarts you lose the count of buckets with quota and the panel in the dashboard becomes wrong right ?

Reply:
The count will stay forever, even if you restart the service (we already follow this logic to count the number of objects)
![image](https://github.com/scality/cloudserver/assets/91462779/3c1f511a-185e-4ebc-ae23-bf9e571b12d0)

To illustrate, I restarted the service after creating the bucket

Reply:
Yes because it got scraped by prometheus.
But now let's stay you got 100 buckets created in a few seconds and before prometheus can scrape (every 30s on artesca) the server restarts, then the 100 increment is lost.

Also the metric should restart from 0 so the counter will go down if prometheus scrape the metric with a value lower than the one before the restart I think

Reply:
In this case the information is lost. What I have in mind, and this is also already done today, is to extend the `reportHandler` (called once every hour) to recompute the up to date metrics, along with a new method in arsenal to count the number of bucket (and accounts) with quota enabled. Then we keep this code as-is, and I add a description that explains the metrics are periodically re-computed.

Reply:
You mean the `reportHandler` will list all buckets to know the up to date number of buckets with quota, number of objects, etc ?

It seems to me this kind of (stateful) counters should maybe not be coming from prometheus but another source like logs form loki

Reply:
Yes, the report handler lists all buckets every hours and compute various statistics.
The loki source cannot be used as loki is currently disabled in Artesca, so this is not reliable.

Reply:
Done in https://github.com/scality/cloudserver/pull/5556/commits/24c9a3ca6df345acb3aaa509f8ee69eae2266805
Arsenal was also updated to increment a counter when the processed bucket has a quota set. See https://github.com/scality/Arsenal/pull/2233/commits/63cc54d3fa3c859773cabeef840fe8aad9e3ccae

Reply:
> But the list doesn't contain all the information, you'll still have to send another request for each bucket to get their config ?

It's an existing code already in prod, we iterate over all the buckets. In a zenko context the number is definitely not millions, but thousands maximum. Anyway millions is also supported, just slower. The change only add a counter, no complexity added.